presto-main/src/main/java/com/facebook/presto/util/MoreMaps.java

```diff
@@ -36,4 +39,18 @@ private MoreMaps() {}
                 .flatMap(Collection::stream)
                 .collect(toMap(Map.Entry::getKey, Map.Entry::getValue, merger));
     }
+
+    public static <K, V> Map<K, V> immutableLinkedMap(Map<K, V> map)
```

Comment:
Because it's not _linked_ (doesn't preserve order)? I'd be happy to drop this completely and use `ImmutableMap.copyOf` everywhere but I'd expect it'd be you, @ArturGajowy, who could object to that :D